pkg/kms/service.go

```diff
@@ -67,3 +74,29 @@ func (s *service) CreateKey(keystoreID string, kt kms.KeyType) (string, error) {
 
 	return keyID, nil
 }
+
+// Sign signs a message using key identified by keyID.
+func (s *service) Sign(keystoreID, keyID string, msg []byte) ([]byte, error) {
+	k, err := s.keystore.Get(keystoreID)
+	if err != nil {
+		return nil, fmt.Errorf(getKeystoreErr, err)
+	}
+
+	if len(k.KeyIDs) == 0 {
+		return nil, errors.New(noKeysErr)
+	}
+
+	for _, id := range k.KeyIDs {
```

Comment:
If first entry in `KeyIDs` is not equal to `keyID`, this will return an error. 

Looping logic here needs update.

Probably create a temporary `kid` variable **before** the loop and in the loop if `id == keyID` then set kid and **after** the loop check:
```if kid == nil { return nil, errors.New(invalidKeyIDErr) }```

Reply:
this was good catch, fixed